core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1393,14 +1393,31 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
     }
   }
 
-  def listSubmissions(workspaceName: WorkspaceName): Future[PerRequestMessage] =
-    dataSource.inTransaction { dataAccess =>
+  def listSubmissions(workspaceName: WorkspaceName): Future[PerRequestMessage] = {
+    val costlessSubmissionsFuture = dataSource.inTransaction { dataAccess =>
       withWorkspaceContextAndPermissions(workspaceName, WorkspaceAccessLevels.Read, dataAccess) { workspaceContext =>
         dataAccess.submissionQuery.listWithSubmitter(workspaceContext)
-          .map(RequestComplete(StatusCodes.OK, _))
       }
     }
 
+    val costMapFuture = costlessSubmissionsFuture flatMap { submissions =>
+      submissionCostService.getWorkflowCosts(submissions.flatMap(_.workflowIds), workspaceName.namespace)
+    }
+
+    toFutureTry(costMapFuture) flatMap {
+      case Failure(ex) =>
+        logger.error("Unable to get cost data from BigQuery", ex)
+        Future.successful(RequestComplete(StatusCodes.OK, costlessSubmissionsFuture))
+      case Success(costMap) =>
+        costlessSubmissionsFuture map { costlessSubmissions =>
+          val costedSubmissions = costlessSubmissions map { costlessSubmission =>
+            costlessSubmission.copy(cost = Some(costlessSubmission.workflowIds.flatMap(id => costMap.get(id).value).sum))
```

Comment:
what does `value` do here?  You should be able to do just `costlessSubmission.workflowIds.flatMap(costMap.get).sum` (note I've also removed the `id =>` and the shorthand `_` since `get` is a single-argument function.

If you wanted to keep using `map`, you could do `costlessSubmission.workflowIds.map(costMap.getOrElse(_,0).sum`

Reply:
🤦‍♂️ This was my hacky way of fixing a bug that was really caused by having `Some().sum`. Going with the first suggestion you made. Thanks for the comment 😄